### PR TITLE
Implement multistage dockerfile for the backend

### PR DESCRIPTION
**Pull request by nullxDEADBEEF**
### Is your feature request related to a problem? Please describe.

Right now there is no difference when we are building the backend in production or in development. Would be awesome to differentiate between the two.

### Describe the solution you'd like

Use multistage in the dockerfile and run based on the target environment.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,25 @@
-FROM tiangolo/uvicorn-gunicorn-fastapi:python3.9-slim
+ARG APP_ENVIRONMENT=dev
+
+
+FROM python:3.9-slim as base
```
Comment (AndreasPB) [inline backend/Dockerfile]: I think there's a problem with the 3.9 image and M1 drivers

```suggestion
FROM --platform=linux/amd64 python:3.9-slim as base
```

This seems to fix it...
Explanation here https://stackoverflow.com/questions/65612411/forcing-docker-to-use-linux-amd64-platform-by-default-on-macos/69636473#69636473

**Review by AndreasPB — COMMENTED** [review]
There was some versioning errors, but I pushed a fix - Test if it works on your PC as well

File: backend/Dockerfile (hunk 1 above)
Comment (AndreasPB) [reply backend/Dockerfile]: I pushed this change

File: .env [hunk 2]
```diff
@@ -1,2 +0,0 @@
-HOST_NAME=localhost
-APP_ENVIRONMENT=dev
```
[the remark this answers is not in the document]
Comment (nullxDEADBEEF) [reply .env:2]: Works on my fine on my machine 👍 

And yeah better update disc ASAP :DD

**Review by AndreasPB — APPROVED** [review]

**Pull request merged**